ice/trace.py

```diff
@@ -370,19 +377,19 @@ def _get_short_list(lst: list, max_length=3) -> list:
     return lst[:max_length] + ["..."] if len(lst) > max_length else lst
 
 
-def _get_first_descendant(value):
-    if value:
-        if isinstance(value, dict):
-            first, *_ = value.values()
-            return _get_first_descendant(first)
-        if isinstance(value, (list, tuple)):
-            if isinstance(value[0], str):
-                return [v for v in value if isinstance(v, str)]
-            return _get_first_descendant(value[0])
-        if hasattr(value, "dict") and callable(value.dict):
-            value = compress(value.dict())
-            return _get_first_descendant(value)
-    return value
+def _get_first_descendant(value: Any) -> Any:
+    if isinstance(value, dict) and value:
+        first, *_ = value.values()
+        return _get_first_descendant(first)
+    elif isinstance(value, (list, tuple)) and value:
+        if isinstance(value[0], str):
+            return [v for v in value if isinstance(v, str)]
+        return _get_first_descendant(value[0])
+    elif hasattr(value, "dict") and callable(value.dict):
+        value = compress(value.dict())
+        return _get_first_descendant(value)
+    else:
+        return value
```

Comment:
I'm surprised the diffs in this file are so large, but don't care much since this is code we want to get rid of soon.

optional: In both get_strings and _get_first_descendant, does it simplify things if we just immediately return if the value is an instance of pd.DataFrame?

Reply:
> I'm surprised the diffs in this file are so large

Ever used github's "hide whitespace" option?

> optional: In both get_strings and _get_first_descendant, does it simplify things if we just immediately return if the value is an instance of pd.DataFrame?

Also need to account for pandas series, numpy arrays, and probably other similar types I don't know about.

Reply:
> Ever used github's "hide whitespace" option?

Yes, and I really wish I could change the default because I don't always remember to turn it on.